File: week6/dfs.java
public class dfs {
    public static boolean[] visited = new boolean[9];
    static int[][] graph = {{}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7}};

    public static void sol(int v) {
        visited[v] = true;

        System.out.print(v + " ");

        for (int node : graph[v]) {
            if (!visited[node]) {
                sol(node);
            }
        }
    }

    public static void main(String[] args) {
	    sol(1);
    }
}

File: week6/dfs.cpp
#include <bits/stdc++.h>

using namespace std;
bool visited[9];

int graph[9][3] = { {}, {2,3,8}, {1,7}, {1,4,5}, {3,5}, {3,4}, {7}, {2,6,8}, {1,7} };

void dfs(int v) {
	visited[v] = true;

	cout << v << ' ';

	for (int node : graph[v]) {
		if (!visited[node]) {
			dfs(node);
		}
	}
}

void main() {
	visited[0] = true;
	dfs(1);
}